haystack/components/rankers/metadata_grouper.py

```diff
@@ -0,0 +1,190 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from collections import defaultdict
+from typing import Any, Dict, List, Optional, Tuple
+
+from haystack import Document, component, logging
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class MetaDataGrouper:
+    """
+    It reorders the documents by grouping them based on metadata keys.
+
+    It groups the documents based on metadata keys. It can group based on a:
+     - single metadata key 'group_by'
+     - further subgroup them based on another metadata key 'subgroup_by'.
+
+    It can also sort the documents within each group or subgroup based on a specified metadata key, 'sort_docs_by'
+
+    It returns a flat list with the original input documents ordered by the 'group_by' and 'subgroup_by' metadata
+    values, and all documents without a group are included at the end of the list.
+
+    This helps to ensure the documents are properly organized leading to a better performance of a following LLM.
+
+    ### Usage example
+
+    ```python
+    from haystack.components.rankers import MetaDataGrouper
+    from haystack.dataclasses import Document
+
+
+    docs = [
+        Document(content="Javascript is a popular programming language", meta={"group": "42", "split_id": 7, "subgroup": "subB"}),
+        Document(content="Python is a popular programming language",meta={"group": "42", "split_id": 4, "subgroup": "subB"}),
+        Document(content="A chromosome is a package of DNA", meta={"group": "314", "split_id": 2, "subgroup": "subC"}),
+        Document(content="An octopus has three hearts", meta={"group": "11", "split_id": 2, "subgroup": "subD"}),
+        Document(content="Java is a popular programming language", meta={"group": "42", "split_id": 3, "subgroup": "subB"})
+    ]
+
+    sample_meta_aggregator = MetaDataGrouper(group_by="group",subgroup_by="subgroup", sort_docs_by="split_id")
+    result = sample_meta_aggregator.run(documents=docs)
+    print(result["documents"])
+    ```
+    """  # noqa: E501
+
+    def __init__(self, group_by: str, subgroup_by: Optional[str] = None, sort_docs_by: Optional[str] = None):
+        """
+        Creates an instance of DeepsetMetadataGrouper.
+
+        :param group_by: The metadata key to aggregate the documents by.
+        :param subgroup_by: The metadata key to aggregate the documents within a group that was created by the
+                            `group_by` key.
+        :param sort_docs_by: Determines which metadata keys are used to sort the documents. If not provided, the
```

Comment:
```suggestion
        :param sort_docs_by: Determines which metadata key is used to sort the documents. If not provided, the
```
Looking at the signature, just one key is supported.